scripts/release-jetty.sh

```diff
@@ -163,6 +163,11 @@ if proceedyn "Are you sure you want to release using above? (y/N)" n; then
     # This is equivalent to 'mvn release:perform'
     if proceedyn "Build/Deploy from tag $TAG_NAME? (Y/n)" y; then
         mvn clean deploy -Peclipse-release $DEPLOY_OPTS
+        echo "IMPORTANT NOTICE: You will need to build+deploy jetty-documentation on JDK21 to make the documentation sane!"
+        echo "Switch to a new window, make sure you are using JDK21, and run the following command:"
+        echo "$ mvn clean deploy -Peclipse-release $DEPLOY_OPTS -pl :jetty-documentation"
```

Comment:
As the jetty-documentation module will have been already deployed by previous steps of the script are we sure re deploying it will work? There is possibly some rule in nexus preventing that. Not sure but worth to check

Reply:
Nope, there is no prevention on oss.sonatype.org on redeploy, we have done that in the past with no issues.

Reply:
Note: this extra step is needed, as if you build `jetty-documentation` on JDK17, you'll get this error (and the documentation will be bad) ...

```
[INFO] Executing: [/home/joakim/java/jvm/jdk-17.0.6+10/bin/java, -Djava.io.tmpdir=/home/joakim/code/jetty/jetty.project-10.0.x/target/tests/test-bases/jetty_base_1846479261761665165/work, -jar, /home/joakim/code/jetty/jetty.project-10.0.x/documentation/jetty-documentation/target/jetty-home-10.0.17-SNAPSHOT/start.jar, --add-modules=threadpool-virtual,http]
[INFO] Working Dir: /home/joakim/code/jetty/jetty.project-10.0.x/target/tests/test-bases/jetty_base_1846479261761665165
[INFO] INFO  : mkdir ${jetty.base}/start.d
[INFO] INFO  : server          transitively enabled, ini template available with --add-module=server
[INFO] INFO  : logging-jetty   transitively enabled
[INFO] INFO  : threadpool-virtual initialized in ${jetty.base}/start.d/threadpool-virtual.ini
[INFO] INFO  : resources       transitively enabled
[INFO] INFO  : http            initialized in ${jetty.base}/start.d/http.ini
[INFO] INFO  : logging/slf4j   transitive provider of logging/slf4j for logging-jetty
[INFO] INFO  : logging/slf4j   dynamic dependency of logging-jetty
[INFO] INFO  : bytebufferpool  transitively enabled, ini template available with --add-module=bytebufferpool
[INFO] INFO  : mkdir ${jetty.base}/resources
[INFO] INFO  : copy ${jetty.home}/modules/logging/jetty/resources/jetty-logging.properties to ${jetty.base}/resources/jetty-logging.properties
[INFO] INFO  : Base directory was modified
[INFO] Executing: [/home/joakim/java/jvm/jdk-17.0.6+10/bin/java, -Djava.io.tmpdir=/home/joakim/code/jetty/jetty.project-10.0.x/target/tests/test-bases/jetty_base_1846479261761665165/work, -jar, /home/joakim/code/jetty/jetty.project-10.0.x/documentation/jetty-documentation/target/jetty-home-10.0.17-SNAPSHOT/start.jar, /home/joakim/code/jetty/jetty.project-10.0.x/documentation/jetty-documentation/target/jetty-home-10.0.17-SNAPSHOT/etc/jetty-halt.xml]
[INFO] Working Dir: /home/joakim/code/jetty/jetty.project-10.0.x/target/tests/test-bases/jetty_base_1846479261761665165
[INFO] 2023-10-06 07:03:27.119:WARN :oejx.XmlConfiguration:main: Config error java.lang.IllegalStateException: No Method: <Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>|          <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>|        </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call> on class java.lang.Thread at <Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>|          <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>|        </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call> in file:///home/joakim/code/jetty/jetty.project-10.0.x/documentation/jetty-documentation/target/jetty-home-10.0.17-SNAPSHOT/etc/jetty-threadpool-virtual.xml
[INFO] 2023-10-06 07:03:27.123:WARN :oejx.XmlConfiguration:main: Config error java.lang.IllegalStateException: No Method: <Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>|          <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>|        </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call> on class java.lang.Thread at <New id="threadPool" class="org.eclipse.jetty.util.thread.QueuedThreadPool"><Set name="name" property="jetty.threadPool.namePrefix"/><Set name="minThreads" type="int"><Property name="jetty.threadPool.minThreads" deprecated="threads.min" default="10"/></Set><Set name="maxThreads" type="int"><Property name="jetty.threadPool.maxThreads" deprecated="threads.max" default="200"/></Set><Set name="reservedThreads" type="int"><Property name="jetty.threadPool.reservedThreads" default="-1"/></Set><Set name="idleTimeout" type="int"><Property name="jetty.threadPool.idleTimeout" deprecated="threads.timeout" default="60000"/></Set><Set name="maxEvictCount" type="int"><Property name="jetty.threadPool.maxEvictCount" default="1"/></Set><Set name="detailedDump" type="boolean"><Property name="jetty.threadPool.detailedDump" default="false"/></Set><Get id="namePrefix" name="name"/><Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>|          <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>|        </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call><Call name="setVirtualThreadsExecutor"><Arg>|        <Call class="java.util.concurrent.Executors" name="newThreadPerTaskExecutor"><Arg><Ref refid="virtualThreadFactory"/></Arg></Call>|      </Arg></Call></New> in file:///home/joakim/code/jetty/jetty.project-10.0.x/documentation/jetty-documentation/target/jetty-home-10.0.17-SNAPSHOT/etc/jetty-threadpool-virtual.xml
[INFO] 2023-10-06 07:03:27.124:WARN :oejx.XmlConfiguration:main: Unable to execute XmlConfiguration
[INFO] java.lang.IllegalStateException: No Method: <Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>|          <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>|        </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call> on class java.lang.Thread
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:958)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:504)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.newObj(XmlConfiguration.java:1028)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:510)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:457)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:359)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1900)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[INFO]  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO]  at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[INFO]  at org.eclipse.jetty.start.Main.invokeMain(Main.java:229)
[INFO]  at org.eclipse.jetty.start.Main.start(Main.java:528)
[INFO]  at org.eclipse.jetty.start.Main.main(Main.java:76)
[INFO] Caused by: 
[INFO] java.lang.NoSuchMethodException: ofVirtual
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:993)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:950)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:504)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.newObj(XmlConfiguration.java:1028)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:510)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:457)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:359)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1900)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[INFO]  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO]  at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[INFO]  at org.eclipse.jetty.start.Main.invokeMain(Main.java:229)
[INFO]  at org.eclipse.jetty.start.Main.start(Main.java:528)
[INFO]  at org.eclipse.jetty.start.Main.main(Main.java:76)
[INFO] java.lang.reflect.InvocationTargetException
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[INFO]  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO]  at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[INFO]  at org.eclipse.jetty.start.Main.invokeMain(Main.java:229)
[INFO]  at org.eclipse.jetty.start.Main.start(Main.java:528)
[INFO]  at org.eclipse.jetty.start.Main.main(Main.java:76)
[INFO] Caused by: java.lang.IllegalStateException: No Method: <Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>
[INFO]           <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>
[INFO]         </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call> on class java.lang.Thread
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:958)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:504)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.newObj(XmlConfiguration.java:1028)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:510)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:457)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:359)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1900)
[INFO]  ... 7 more
[INFO] Caused by: java.lang.NoSuchMethodException: ofVirtual
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:993)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:950)
[INFO]  ... 13 more
[INFO] java.lang.reflect.InvocationTargetException
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO]  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[INFO]  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO]  at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[INFO]  at org.eclipse.jetty.start.Main.invokeMain(Main.java:229)
[INFO]  at org.eclipse.jetty.start.Main.start(Main.java:528)
[INFO]  at org.eclipse.jetty.start.Main.main(Main.java:76)
[INFO] Caused by: java.lang.IllegalStateException: No Method: <Call class="java.lang.Thread" name="ofVirtual"><Call class="java.lang.Thread$Builder" name="name"><Arg>
[INFO]           <Property name="jetty.threadPool.virtual.namePrefix"><Default><Ref refid="namePrefix"/>-virtual-</Default></Property>
[INFO]         </Arg><Arg type="long">0</Arg><Call class="java.lang.Thread$Builder" name="inheritInheritableThreadLocals"><Arg type="boolean"><Property name="jetty.threadPool.virtual.inheritInheritableThreadLocals" default="false"/></Arg><Call id="virtualThreadFactory" class="java.lang.Thread$Builder" name="factory"/></Call></Call></Call> on class java.lang.Thread
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:958)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:504)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.newObj(XmlConfiguration.java:1028)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:510)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:457)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:359)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1900)
[INFO]  ... 7 more
[INFO] Caused by: java.lang.NoSuchMethodException: ofVirtual
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:993)
[INFO]  at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:950)
[INFO]  ... 13 more
[INFO] 
[INFO] Usage: java -jar $JETTY_HOME/start.jar [options] [properties] [configs]
[INFO]        java -jar $JETTY_HOME/start.jar --help  # for more information
```